src/protocol.h

```diff
@@ -277,6 +278,7 @@ enum ServiceFlags : uint64_t {
     // BIP process.
 };
 
+extern std::atomic<bool> g_initial_block_download_completed;
```

Comment:
I don't like that we need to introduce a global here, especially one that has to be accessible externally (in the header file).
Any drawback to moving `GetDesirableServiceFlags` to the .cpp instead?